src/plugin.cc

```diff
@@ -306,32 +312,43 @@ void plugin::Zeek_Spicy::Plugin::registerPacketAnalyzer(const std::string& name,
     _packet_analyzers_by_type[info.type] = info;
 }
 
-void plugin::Zeek_Spicy::Plugin::registerEnumType(
-    const std::string& ns, const std::string& id,
-    const hilti::rt::Vector<std::tuple<std::string, hilti::rt::integer::safe<int64_t>>>& labels) {
-    if ( ::zeek::detail::lookup_ID(id.c_str(), ns.c_str()) )
+hilti::Result<hilti::Nothing> plugin::Zeek_Spicy::Plugin::registerType(const hilti::ID& id) {
+    auto ti = _driver->lookupType(id);
+    if ( ! ti )
+        return hilti::result::Error("unknown type");
+
+    auto type = createZeekType(ti->type, id);
+    if ( ! type )
+        return type.error();
+
+    registerType(ti->id, *type);
+    return hilti::Nothing();
+}
+
+void plugin::Zeek_Spicy::Plugin::registerType(const hilti::ID& id, const ::zeek::TypePtr& type) {
+    if ( ::zeek::detail::lookup_ID(id.str().c_str(), id.namespace_().str().c_str()) ) {
```

Comment:
Since we now provide users an explicit way to inject input into this method, I wonder whether we shouldn't instead throw an error if the same `id` is registered multiple times (maybe check whether `type` is inconsistent?). The defensive pattern would be to always first call `findType` and only if it finds nothing to register a new type.

Reply:
Just noticed the debug statement added in a later patch. Maybe that is good enough, but I suspect you added this because you ran into exactly the issue a hard error had prevented.